### PR TITLE
Explain confirmation via trusted names more clearly

### LINKED ISSUE #96: Mutual Confirmation "trustedNames" value requirements
The document currently says:
- The JSON dictionary MAY contain a key "trustedNames" that is an array of strings containing domains that can be used for mutual confirmation of resolver designation.

This is ambiguous with regards to what valid values are, and more importantly, what logic a client should use to reject values. Otherwise, a resolver could list friendly servers as trusted and neither could have a real relationship with the dnsZones.

I think this may be as simple as being clear about trustedNames needing to be identical to or parent zones of the dnsZones the resolver is claiming. However, this does have an interesting edge case where the resolver claims several dnsZones which have no common parent other than the root zone.

It's also likely I'm missing some other qualifier intended by saying "can be used for mutual confirmation" but whatever that is should be listed explicitly in the document.

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #96

### CODE REVIEW

File: draft-pauly-add-resolver-discovery.md [hunk 1]
```diff
@@ -218,18 +230,18 @@ Note that the domains listed in "trustedNames" may be broader than the zones tha
    }
 ~~~
 
-# Confirmation of Domain Name Ownership with TLS Certificates {#cert-name-check}
+## Confirmation of Designation with TLS Certificates {#confirm-cert-name}
 
-A DoH server designation without DNSSEC or a PvD for mutual confirmation can be confirmed by the 
-SubjectAlternativeName field in the TLS certificate. When a client wants to confirm the validity of the 
-designation in this situation, it checks the TLS certificate of the DoH server for the name of the domain 
+A DoH server designation can also be validating by checking the SubjectAlternativeName field in the DoH
```
Comment (ekinnear) [inline draft-pauly-add-resolver-discovery.md:235]: Nit: “validating” -> “validated”

**Comment by mstojens** [comment]
Agree with the discovered nits, though I wouldn't block IETF submission on them. Looks great, and thanks for addressing 96!

File: draft-pauly-add-resolver-discovery.md [hunk 2]
```diff
@@ -103,18 +103,18 @@ equivalent to querying a particular Direct Resolver.
...
 
-There are several methods that can be used to designate a resolver:
+There are several methods that can be used to discover and validate a resolver designation:
 
-- Based on SVCB DNS records issued to another resolver ({{svcb}})
-- Based on information in a provisioning domain (PvD) from the Designated DoH Resolver that is confirmed via SVCB DNS records ({{pvd}})
-- Based on mutual agreement through confirmation of domains over HTTPS ({{pvd-mutual}})
-- Based on confirmation of domain name ownership within TLS certificates ({{cert-name-check}})
+- Discovery using SVCB DNS records ({{svcb}}), and validation using DNSSEC
+- Discovery using information in a provisioning domain (PvD) file from the Designated DoH Resolver
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-pauly-add-resolver-discovery.md]: Fixed!

File: draft-pauly-add-resolver-discovery.md [hunk 3]
```diff
@@ -166,13 +166,25 @@ The "expires" key indicates a time after which the content of the PvD file is no
...
+## Confirmation of Designation with Zone Apex PvD {#confirm-zone-apex}
 
-Designated DoH Resolvers that provide the PvD JSON described in {{pvd}} can also provide information to allow validation of zone designations without DNSSEC.
+Designated DoH Resolvers that provide the PvD JSON described in {{pvd}} can also provide information to validate of zone's designation without DNSSEC.
+In order to confirm the designation, the client requests a well-known HTTPS URI based on a zone apex name, and checks a PvD file to ensure that
+it matches the DoH resolver. This ensures that a DoH resolver cannot claim a designation for a given zone without cooperation from the entity that owns
+the certificate for the apex of that zone.
 
-The JSON dictionary MAY contain a key "trustedNames" that is an array of strings containing domains that can be used for mutual confirmation of resolver designation.
+In order to list out the zone apex names that confirm designation in this manner, the DoH resolver's PvD JSON dictionary can contain
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-pauly-add-resolver-discovery.md]: Fixed!

File: draft-pauly-add-resolver-discovery.md [hunk 4]
```diff
@@ -151,9 +151,9 @@ Designated Resolvers that support DoH SHOULD provide a PvD JSON dictionary avail
...
+Names that are listed in the "dnsZones" key in the JSON dictionary indicate a set of zones that designate the resolver. These are the zones that are available to resolve through the associated DoH server. Note that this list does not need to be exhaustive, but is the set of common zones managed by the resolver that all clients should be aware of. Before using  DNS results for these names, clients MUST validate the designation either with a DNSSEC-signed SVCB record ({{svcb}}), or the confirmation methods described in {{confirm-zone-apex}} and {{confirm-cert-name}}. DNS queries for validating records SHOULD be sent the DoH resolver. In order to optimize the validation of these domains, servers MAY use HTTP Server Push to deliver the signed SVCB answers prior to requests being made.
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-pauly-add-resolver-discovery.md]: I don't see the double space in the actual file? But fixed the nit too =)

File: draft-pauly-add-resolver-discovery.md (hunk 1 above)
Comment (tfpauly) [reply draft-pauly-add-resolver-discovery.md]: Fixed

File: draft-pauly-add-resolver-discovery.md [hunk 5]
```diff
@@ -166,13 +166,25 @@ The "expires" key indicates a time after which the content of the PvD file is no
...
+In order to confirm the designation, the client requests a well-known HTTPS URI based on a zone apex name, and checks a PvD file to ensure that
+it matches the DoH resolver. This ensures that a DoH resolver cannot claim a designation for a given zone without cooperation from the entity that owns
+the certificate for the apex of that zone.
 
-The JSON dictionary MAY contain a key "trustedNames" that is an array of strings containing domains that can be used for mutual confirmation of resolver designation.
+In order to list out the zone apex names that confirm designation in this manner, the DoH resolver's PvD JSON dictionary can contain
+an array of strings, with the key "trustedNames". Clients can validate the resolver designation by checking a resource hosted by a
+name indicated in "trustedNames". The client first issues an HTTP GET request by appending "/.well-known/pvd" to the trusted name,
+using the "https" scheme. The client's query for the address of the trusted name MAY use the DoH resolver prior to fully validating the designation.
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-pauly-add-resolver-discovery.md]: Added a clause to the sentence to explain why this is safe—it's what you assumed.

[the remark this answers is not in the document]
Comment (ekinnear) [reply draft-pauly-add-resolver-discovery.md]: Nice, thank you!

**Pull request merged**